acceptance/tests/direct_puppet/supports_utf8.rb

```diff
@@ -5,14 +5,20 @@
   app_type = File.basename(__FILE__, '.*')
   tmp_environment   = mk_tmp_environment(app_type)
 
-  filenames         = {'windows' => "C:/cygwin64/tmp/#{tmp_environment}_file"}
-  filenames.default = "/tmp/#{tmp_environment}_file"
+  tmp_file = {}
+  agents.each do |agent|
```

Comment:
You can't just ask for the first Windows agent and the first non-Windows agent to avoid the loop?